include/SFML/Window/Event.hpp

```diff
@@ -34,196 +34,298 @@
-        float        x;    //!< Current value of the sensor on X axis
-        float        y;    //!< Current value of the sensor on Y axis
-        float        z;    //!< Current value of the sensor on Z axis
+        Sensor::Type type{}; //!< Type of the sensor
+        float        x{};    //!< Current value of the sensor on X axis
+        float        y{};    //!< Current value of the sensor on Y axis
+        float        z{};    //!< Current value of the sensor on Z axis
     };
 
     ////////////////////////////////////////////////////////////
     /// \brief Enumeration of the different types of events
     ///
+    /// Each event type maps 1-to-1 to a specific structure of
+    /// the same name.
+    ///
     ////////////////////////////////////////////////////////////
-    enum EventType
+    enum class Type
     {
-        Closed,                 //!< The window requested to be closed (no data)
-        Resized,                //!< The window was resized (data in event.size)
-        LostFocus,              //!< The window lost the focus (no data)
-        GainedFocus,            //!< The window gained the focus (no data)
-        TextEntered,            //!< A character was entered (data in event.text)
-        KeyPressed,             //!< A key was pressed (data in event.key)
-        KeyReleased,            //!< A key was released (data in event.key)
-        MouseWheelScrolled,     //!< The mouse wheel was scrolled (data in event.mouseWheelScroll)
-        MouseButtonPressed,     //!< A mouse button was pressed (data in event.mouseButton)
-        MouseButtonReleased,    //!< A mouse button was released (data in event.mouseButton)
-        MouseMoved,             //!< The mouse cursor moved (data in event.mouseMove)
-        MouseEntered,           //!< The mouse cursor entered the area of the window (no data)
-        MouseLeft,              //!< The mouse cursor left the area of the window (no data)
-        JoystickButtonPressed,  //!< A joystick button was pressed (data in event.joystickButton)
-        JoystickButtonReleased, //!< A joystick button was released (data in event.joystickButton)
-        JoystickMoved,          //!< The joystick moved along an axis (data in event.joystickMove)
-        JoystickConnected,      //!< A joystick was connected (data in event.joystickConnect)
-        JoystickDisconnected,   //!< A joystick was disconnected (data in event.joystickConnect)
-        TouchBegan,             //!< A touch event began (data in event.touch)
-        TouchMoved,             //!< A touch moved (data in event.touch)
-        TouchEnded,             //!< A touch event ended (data in event.touch)
-        SensorChanged,          //!< A sensor value changed (data in event.sensor)
-
-        Count //!< Keep last -- the total number of event types
+        Empty,                  //!< Default type
+        Closed,                 //!< The window requested to be closed
```

Comment:
Since we're throwing around the whole event structure, I'd really love to change `closed`. I remember this confusing me from day one, since it doesn't signal the window is gone, it only signals the someone or something wanting to close the window. I'm just not sure about a new name. Not a real fan of it, but maybe something along the lines of `WindowCloseRequests`? (Also see next comment.)